src/onegov/org/views/newsletter.py

```diff
@@ -80,13 +80,62 @@ def get_newsletter_form(
     return form
 
 
-def news_by_newsletter(
+def newsletter_news_by_access(
     newsletter: Newsletter,
-    request: 'OrgRequest'
+    request: 'OrgRequest',
+    access: str = 'public',
+    include_wider_access: bool = True,
 ) -> list[News] | None:
+    """
+    Retrieves a list of news items associated with a specific newsletter
+    based on the given access level or higher. This function is used for the
+    newsletter preview as well as for sending newsletters.
+
+    The function filters the news items based on their access level ('public',
+    'secret', 'private') and whether they are published.
+    It then orders the news items in descending order by their creation date.
+
+    Returns:
+    - list[News] | None: A list of news items that match the provided access
+    level. If no news items are found, None is returned.
+
+    Raises:
+    - ValueError: If an invalid access level is provided.
+    """
 
     news_ids = newsletter.content.get('news')
+    if not news_ids:
+        return None
+
+    if access not in ('public', 'secret', 'private'):
```

Comment:
```suggestion
    if access not in ORDERED_ACCESS:
```
There's no reason to restrict this function to only those three access levels